src/App.tsx

```diff
@@ -1,34 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { filterTodos } from './utils/filterTodos';
+import { FilterByState } from './types/FilterByState';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState<FilterByState>(FilterByState.ALL);
+  const [query, setQuery] = useState('');
+  const [hasErrorMessage, setErrorMessage] = useState('');
```

Comment:
The state variable `hasErrorMessage` is misleadingly named. Since it holds a string message rather than a boolean, consider renaming it to `errorMessage` to better reflect its purpose.